osuosl/climesync
# Automatic reauth on token expiration

fixes issue #62 

## Changes in this PR.


- [X] Checks if the user's token is expired before performing a command
- [X] If the user is logged in as the same user and to the same server as is in the config, re-authentication is performed silently
- [X] If the user's credentials don't match what is in the config, an error message is shown asking the user to sign in again before continuing

## Testing this PR.


1. Run `nosetests`
2. Edit `util.py` and change the `<=` on line 81 to `>` (So you don't have to wait 30 minutes for your token to expire)
3. Run climesync, make sure that you're connected to the timesync-staging server, and sign in as `admin` with the password `timesync-staging`. When asked if you want to save the credentials to the config file, answer yes (If they're already in the config file and you're not asked, that works too)
4. Attempt to run `gt` and see that the command works with no indication that you were signed back in
5. Sign in as `test` with the password `test` (Be sure to answer `n` when asked if you want to save to the config file)
6. Attempt to run `gt` again and see that you're told to sign in again

@osuosl/devs
